.github/workflows/lint_python.yml

```diff
@@ -0,0 +1,18 @@
+name: lint_python
+on:
+  pull_request:
+  push:
+  #  branches: [master]
+jobs:
+  lint_python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@master
+      - run: pip install black codespell flake8 isort pytest
+      - run: black . --diff || true
+      - run: codespell --quiet-level=2 || true  # --ignore-words-list="" --skip=""
+      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: isort --recursive . || true
```

Comment:
Thanks for adding all these linters. Would it be possible for you to also fix the lint errors so that we can avoid the `|| true` which silences the errors?

Reply:
I find that PRs that do lots of things and touch many files are difficult to review and the more months that pass before they are merged, the more rebasing and other rework that is required.  It is better to do small, focused PRs that put tests in place and run them in _allow failures_ mode so that it is clear what future PRs need to achieve to remove the _allow failures_ mode for each individual test.  If we made all tests mandatory in a single PR then it would be complicated to review and would take months to land.

Reply:
That's fair enough. I find though that on most projects people aren't going to check the logs of the CI run when there's a green tick next to it. As such, I would suggest to use `if: always()` on the step instead of `|| true` in the command line to not break the build on errors.